hack/kind/kind.sh

```diff
@@ -1,18 +1,48 @@
 #!/bin/sh
 
-set -e
+set -ex
+info_message() {
+    if [ -z "${1}" ]; then
+        echo "info_message() requires a message"
+        exit 1
+    fi
+    echo -e "[\033[0;34m ACTION \033[0m] ${1}"
+}
+
+pass_message() {
+    if [ -z "${1}" ]; then
+        echo "pass_message() requires a message"
+        exit 1
+    fi
+    echo -e "[\033[0;32m PASSED \033[0m] ${1}"
+}
+
+error_message() {
+    if [ -z "${1}" ]; then
+        echo "error_message() requires a message"
+        exit 1
+    fi
+    if [ -n "$1" ]; then
+        echo -e "[\033[0;31m FAILED \033[0m] ] ${1}"
+    fi
+}
 
 up() {
+    if ! [ -x "$(command -v kubectl)" ]; then
+        error_message "kubectl is required. Please install kubectl"
+        exit 1
+    fi
```

Comment:
Might as well check for `kind` while you're at it

Reply:
I added that initially, but removed it, because kind itself does that. It throws `kind not found`.

Reply:
My above comment is hilarious 🗡️ , `Kind` throws `'kind not found` :). I meant it's the first command for up and down both and shell throws kind not found if it doesn't exist, so i thought it's redundant. But no harm in  explicit check.

Reply:
@russellb added explicit check for kind.